python/cudf/cudf/dataframe/categorical.py

```diff
@@ -224,13 +251,10 @@ def _unique_segments(self):
         return segs, sortedvals
 
     def unique(self, method=None):
-        return CategoricalColumn(
-            data=Buffer(
-                list(range(0, len(self._categories))), categorical=True
-            ),
-            categories=self._categories,
-            ordered=self._ordered,
-        )
+        codes = Buffer(cudautils.arange(len(self._categories)))
```

Comment:
This assumes that every code is in the Category. You need to just call `.unique()` against the codes and use that.

Reply:
Updated :+1: